common/cuda_hip/matrix/batch_csr_kernels.hpp.inc

```diff
@@ -111,3 +111,47 @@ __global__ __launch_bounds__(
                        x_b.values);
     }
 }
+
+
+template <typename ValueType, typename IndexType>
+__device__ __forceinline__ void scale(
+    const int num_rows, const ValueType* const __restrict__ col_scale,
+    const ValueType* const __restrict__ row_scale,
+    const IndexType* const __restrict__ col_idxs,
+    const IndexType* const __restrict__ row_ptrs,
+    ValueType* const __restrict__ values)
+{
+    constexpr auto warp_size = config::warp_size;
+    const auto tile =
+        group::tiled_partition<warp_size>(group::this_thread_block());
+    const int tile_rank = threadIdx.x / warp_size;
+    const int num_tiles = (blockDim.x - 1) / warp_size + 1;
```

Comment:
I don't understand what this is. From the name, `blockDim.x / warp_size` makes much more sense. For the default block size, this is basically `(blockDim / warp_size) - 1`, but if you were to use `blockDim = 512` then you would have `(blockDIm / warp_size) - 2`, which seems inconsistent to me.

Reply:
Sorry, from the name `num_tiles = ceildiv(blockDim.x, warp_size)` makes more sense.